Add description and examples for SQL Keywords (#25729)
DataDog/documentation: content/en/ddsql_editor/reference/expressions_and_operators.md

For tag references and tag groups, the equality operator (`=`) is treated as a "contains" comparison. See the [Querying Tags in DDSQL][1] for more details.

## SQL comparison keywords

> DDSQL supports the following SQL keywords, which function as standard boolean operators:

Author: ```suggestion
DDSQL supports the following SQL keywords, which function as standard Boolean operators:
```

| Operator | Description            | Example | Result |
|----------|------------------------|---------|--------|

> | `NOT`    | Filter records based on more than one condition. | `SELECT * FROM Student WHERE NOT Major = 'Economics';`   | Return all students that do not major in Economics.  |

Reviewer: Can we generally replace these examples by examples relevant in the context of Datadog? I'll give you a few
Here, use 
select * from host where not env = 'prod'

> | `AND`    | Filter records based on more than one condition. | `SELECT * FROM Student WHERE Age = 18 AND Major = 'Economics';`   | Return all students that are age 18 AND major in Economics.  |

Reviewer: select * from host where env = 'prod'and cloud_provider = 'aws'

> | `OR`     | Filter records based on more than one condition. | `SELECT * FROM Student WHERE Age = 18 OR Major = 'Economics';`   | Return all students that are either age 18 OR major in Economics.  |

Reviewer: select * from host where env = 'prod' or cloud_provider = 'aws'

DDSQL also supports the following comparator keywords as they are defined in the SQL standard:

| Operator     | Description            | Example | Result |
|--------------|------------------------|---------|--------|

> | `IS NULL`    | Select rows if the specified field is null. | `SELECT * FROM albums WHERE artist IS NULL`   | Return all rows that contain no data in the `artist` column.  |

Reviewer: select *
from host
where cloud_provider IS NULL

Author: ```suggestion
| `IS NULL`    | Select rows if the specified field is null. | `SELECT * FROM host WHERE cloud_provider IS NULL`   | Return all rows that contain no data in the `cloud_provider` column.  |
```

> | `IS NULL`    | Select rows if the specified field is null. | `SELECT * FROM host WHERE cloud_provider IS NULL`   | Return all rows that contain no data in the `cloud_provider` column.  |

Author: ```suggestion
| `IS NULL`    | Select rows if the specified field is null. | `SELECT * FROM host WHERE cloud_provider IS NULL;`   | Return all rows that contain no data in the `cloud_provider` column.  |
```

> | `IS NOT NULL`| Select rows if the specified field is not null. Exclude rows with missing data. | SELECT * FROM albums WHERE artist IS NOT NULL | Return all rows that contain data in the `artist` column.   |

Reviewer: select *
from host
where cloud_provider IS NOT NULL

Author: ```suggestion
| `IS NOT NULL`| Select rows if the specified field is not null. Exclude rows with missing data. | SELECT * FROM host WHERE cloud_provider IS NOT NULL | Return all rows that contain data in the `cloud_provider` column.   |
```

> | `IS NOT NULL`| Select rows if the specified field is not null. Exclude rows with missing data. | SELECT * FROM host WHERE cloud_provider IS NOT NULL | Return all rows that contain data in the `cloud_provider` column.   |

Author: ```suggestion
| `IS NOT NULL`| Select rows if the specified field is not null. Exclude rows with missing data. | `SELECT * FROM host WHERE cloud_provider IS NOT NULL;` | Return all rows that contain data in the `cloud_provider` column.   |
```

> | `LIKE`       | Search for a specific pattern in a string value. You can use the following wildcard characters to define the patterns: <br>**Percent sign (%)**: Represents zero, one, or multiple characters. <br>**Underscore (_)**: Represents a single character. | `SELECT * FROM employees WHERE firstname LIKE 'a%';` | Return all rows from the `employees` table where the `firstname` column value starts with `a`.  |

Author: ```suggestion
| `LIKE`       | Search for a specific pattern in a string value. You can use the following wildcard characters to define the patterns: <br>**Percent sign (%)**: Represents zero, one, or multiple characters. <br>**Underscore (_)**: Represents a single character. | `SELECT * FROM aws_eks_cluster WHERE LOWER(logging) LIKE '%"enabled":true%' | Return all rows from the `aws_eks_cluster` table where the `logging` column is `"enabled":true`.  |
```

> | `NOT LIKE`   | Exclude rows from a search, where the row has a specific pattern in a string value. You can use the wildcards `%` and `_` for pattern matching. | `SELECT * FROM products WHERE product_code NOT LIKE '_B%';` | Return all rows from the `products` table where the `product_code` does **not** have `B` as the second character. |

Reviewer: SELECT *
FROM aws_eks_cluster
WHERE LOWER(logging) NOT LIKE '%"enabled":true%'

Author: ```suggestion
| `NOT LIKE`   | Exclude rows from a search, where the row has a specific pattern in a string value. You can use the wildcards `%` and `_` for pattern matching. | `SELECT * FROM aws_eks_cluster WHERE WHERE LOWER(logging) NOT LIKE '%"enabled":true%'` | Return all rows from the `aws_eks_cluster` table where the `logging` does **not** have `"enabled":true%'`. |
```

> | `IN`         | Find multiple values in a `WHERE` clause. The `IN` operator is shorthand for multiple `OR` conditions. | `SELECT * FROM Customers WHERE Country IN ('Germany', 'France', 'UK');`  | Return all rows from `Customers` table where the `Country` value is either 'Germany', 'France', or 'UK'.|

Reviewer: select *
from host
where cloud_provider IN ('aws', 'gcp')

Author: ```suggestion
| `IN`         | Find multiple values in a `WHERE` clause. The `IN` operator is shorthand for multiple `OR` conditions. | `SELECT * FROM host WHERE cloud_provider IN ('aws', 'gcp')`  | Return all rows from `host` table where the `cloud_provider` value is either 'aws', or 'gcp'.|
```

> | `NOT IN`     | Replace a set of arguments with the `<>` or `!=` operator that is combined with the `AND` operator| `SELECT Name FROM Emp WHERE age NOT IN (23, 22, 21);`  | Return the `Names` of the `Employees` who do not have `Age` equal to `23`, `22`, or `21`. |

Reviewer: select *
from host
where cloud_provider NOT IN ('aws', 'gcp')

Author: ```suggestion
| `NOT IN`     | Replace a set of arguments with the `<>` or `!=` operator that is combined with the `AND` operator| `SELECT * FROM host WHERE cloud_provider NOT IN ('aws', 'gcp');`  | Return all rows where `cloud_provider` is not `aws` or `gcp`. |
```
